app/controllers/api_controller.rb

```diff
@@ -0,0 +1,8 @@
+class ApiController < ApplicationController
+  skip_before_action :verify_authenticity_token
+
+  private
+  def authenticated?
+    authenticate_or_request_with_http_basic {|username, password_digest| User.where( username: username, password_digest: password_digest).present? }
```

Comment:
`User.where` will query the database for all records that may match these conditions, pull them in to memory, then you check if it exists use `present?`.  Instead I would use [exists?](http://devdocs.io/rails/activerecord/findermethods#method-i-exists-3F) which will perform slightly better as it only queries the database to check if any records matching these criteria exists.